src/cryptonote_core/blockchain.cpp

```diff
@@ -576,6 +576,26 @@ bool Blockchain::deinit()
   return true;
 }
 //------------------------------------------------------------------
+// This function removes blocks from the top of blockchain.
+// It starts a batch and calls private method pop_block_from_blockchain().
+void Blockchain::pop_blocks(uint64_t nblocks)
+{
+  CRITICAL_REGION_LOCAL(m_tx_pool);
+  CRITICAL_REGION_LOCAL1(m_blockchain_lock);
+
+  while (!m_db->batch_start())
+  {
+    epee::misc_utils::sleep_no_w(1000);
+  }
+
+  for (uint64_t i=0; i < nblocks; ++i)
+  {
+    pop_block_from_blockchain();
+  }
```

Comment:
I don't think popping will normally throw, but it's safer to try so the batch can be stopped below in case it does.